### PR TITLE
feat(rule): add subtype CDM_CODE verification to regional waste classification

### PR DESCRIPTION
**Pull request by AMarcosCastelo**
### 🧾 Summary

Adds subtype CDM_CODE verification to the regional waste classification rule and refactors test cases to reduce code duplication.

### 🧩 Context

The regional waste classification rule needs to verify that the provided subtype matches the CDM_CODE for the given Local Waste Classification ID. This ensures data consistency and prevents mismatched waste classification data. Additionally, the test cases file had significant code duplication that was reduced through helper functions.

### 🧱 Scope of this PR

**Included:**
- Added subtype-to-CDM_CODE mapping verification in the regional waste classification processor
- Extracted helper functions (`getCdmCodeFromSubtype`, `normalizeClassificationId`, `normalizeDescriptionForComparison`) to a new helpers file
- Moved `SUBTYPE_TO_CDM_CODE_MAP` to the constants file
- Updated `EFB_SIMILAR_TO_GARDEN_YARD_AND_PARK_WASTE` mapping to '8.7A'
- Temporarily removed `TOBACCO` subtype from the mapping
- Refactored test cases to use helper functions, significantly reducing duplication
- Added test cases for missing subtype and invalid subtype mapping scenarios

**Not included:**
- Other waste classification rules
- Additional subtype mappings beyond what's specified

### 🧪 How to test

Run the test suite for the regional waste classification rule:

```bash
pnpm test methodologies-bold-rule-processors-mass-id-regional-waste-classification
```

All 28 tests should pass with 100% code coverage. The tests verify:
- Valid classifications with matching subtypes
- Invalid subtype mappings
- Missing subtype handling
- CDM_CODE mismatches
- Edge cases with normalized IDs and descriptions

### 🧠 Considerations for Review

- **Type Safety**: The code uses TypeScript's `exactOptionalPropertyTypes`, which requires careful handling of `undefined` values in test cases
- **Helper Functions**: Helper functions were extracted to improve code organization and reusability
- **Test Refactoring**: Test cases were significantly refactored to use helper functions, reducing duplication from ~400 lines to a more maintainable structure
- **Subtype Requirement**: The `subtype` field is now required (validated by another rule), so the processor assumes it's always present
- **Map Lookup**: Changed from multiple `if` statements to a `Map` lookup for better performance and maintainability

### 🔗 Related Links

N/A

---

- [x] **I, the PR author, confirm that this PR works as expected and does not break any service. I also confirm that I applied best practices and improved the codebase quality.**

## Summary by CodeRabbit

* **New Features**
  * Support for waste subtypes with automated CDM code lookup and stricter validation.
  * Improved normalization of classification IDs and descriptions for reliable comparisons.

* **Bug Fixes**
  * New error states for missing subtype mappings and CDM-code mismatches; clearer validation flow.

* **Tests**
  * Expanded E2E/unit tests to cover subtype scenarios, partial-document injection, and edge cases.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

---

> [!NOTE]
> Adds subtype-based CDM_CODE validation to the regional waste classification rule and refactors utilities/tests for clarity.
> 
> - Validates `document.subtype` via `SUBTYPE_TO_CDM_CODE_MAP` and ensures its CDM code matches the IBAMA entry for the provided `LOCAL_WASTE_CLASSIFICATION_ID` and description; returns specific failures for ID/description mismatch, subtype mismatch, unmapped subtype, and non-BR recycler
> - Extracts helpers (`getCdmCodeFromSubtype`, `normalizeClassificationId`, `normalizeDescriptionForComparison`) and moves subtype map to constants (incl. `EFB_SIMILAR_TO_GARDEN_YARD_AND_PARK_WASTE` → `8.7A`)
> - Updates processor to use helpers; subject now includes `subtype` and rule is not applicable when absent
> - Expands unit/E2E tests to pass `partialDocument.subtype`, adding cases for CDM_CODE mismatches, unmapped/missing subtype, and normalized IDs/descriptions
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 0a4bae3ae850c7cb0c69abddedf514e050efb6ed. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: libs/methodologies/bold/rule-processors/mass-id/regional-waste-classification/src/regional-waste-classification.constants.ts [hunk 1]
```diff
@@ -923,3 +925,18 @@ export const WASTE_CLASSIFICATION_CODES = {
     },
   },
 } as const;
+
+/**
+ * Maps MassIdOrganicSubtype values to their corresponding CDM_CODE values.
```
Comment (cris-santos) [inline libs/methodologies/bold/rule-processors/mass-id/regional-waste-classification/src/regional-waste-classification.constants.ts:930]: same comment made in the other PR regarding MassID naming convetion

File: libs/methodologies/bold/rule-processors/mass-id/regional-waste-classification/src/regional-waste-classification.processor.ts [hunk 2]
```diff
@@ -37,6 +40,8 @@ export const RESULT_COMMENTS = {
   CLASSIFICATION_ID_MISSING: `The "${LOCAL_WASTE_CLASSIFICATION_ID}" was not provided.`,
   INVALID_CLASSIFICATION_DESCRIPTION: `The "${LOCAL_WASTE_CLASSIFICATION_DESCRIPTION}" does not match the expected IBAMA code description.`,
   INVALID_CLASSIFICATION_ID: `The "${LOCAL_WASTE_CLASSIFICATION_ID}" does not match an IBAMA code accepted by the methodology.`,
+  INVALID_SUBTYPE_CDM_CODE_MISMATCH: `The subtype does not match the CDM_CODE for the provided Local Waste Classification ID.`,
+  INVALID_SUBTYPE_MAPPING: `The provided subtype does not map to a valid CDM_CODE.`,
```
Comment (cris-santos) [inline libs/methodologies/bold/rule-processors/mass-id/regional-waste-classification/src/regional-waste-classification.processor.ts]: Shouldn't it be `CDM code` instead of `CDM_CODE`?

**Review by cris-santos — DISMISSED** [review]

File: libs/methodologies/bold/rule-processors/mass-id/regional-waste-classification/src/regional-waste-classification.processor.ts (hunk 2 above)
Comment (AMarcosCastelo) [reply libs/methodologies/bold/rule-processors/mass-id/regional-waste-classification/src/regional-waste-classification.processor.ts]: Good Point

Comment (AMarcosCastelo) [reply libs/methodologies/bold/rule-processors/mass-id/regional-waste-classification/src/regional-waste-classification.processor.ts]: [style(rule): use constants in regional waste classification error mes…](https://github.com/carrot-foundation/methodology-rules/pull/328/commits/0a4bae3ae850c7cb0c69abddedf514e050efb6ed)
